### PR TITLE
Publicize: Rename Publicize to Jetpack Social

### PR DESCRIPTION
**Pull request by danielpost**
This PR replaces all user-facing instances of Publicize with Jetpack Social.

#### Changes proposed in this Pull Request:

* Replace Publicize with Jetpack Social.
* If applicable, replace Publicize with "share" (when Publicize is used as a verb).

#### Other information:

- [X] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

pdrWKz-vt-p2

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Make sure tests still pass.
* In general, since they're just textual changes, go through the code and make sure nothing has changed beyond just textual changes.
* Make sure the name is updated in the visible places:

image

image

### CODE REVIEW

**Comment by spsiddarthan** [comment]
This is looking good, we need to rebase both this PR and the phab patch!

File: docs/development-environment.md [hunk 1]
```diff
@@ -305,7 +305,7 @@ To execute them in your local environment, you can use the following commands.
 
 We strongly recommend that you install tools to review your code in your IDE. It will make it easier for you to notice any missing documentation or coding standards you should respect. Most IDEs display warnings and notices inside the editor, making it even easier.
 
-- Jetpack's custom Code Sniffer ruleset is located at `./projects/packages/codesniffer/Jetpack/ruleset.xml`. Depending on your IDE, you can use this path or you may need to use `.phpcs.xml.dist` in the monorepo root. 
+- Jetpack's custom Code Sniffer ruleset is located at `./projects/packages/codesniffer/Jetpack/ruleset.xml`. Depending on your IDE, you can use this path or you may need to use `.phpcs.xml.dist` in the monorepo root.
```
Comment (spsiddarthan) [inline docs/development-environment.md:312]: Was this an unintended change?

Comment (danielpost) [reply docs/development-environment.md:312]: My IDE automatically removes trailing spaces. I figured it's fine to keep the change here 😄

**Review by spsiddarthan — APPROVED** [review]

**Review by bindlegirl — APPROVED** [review]
LGTM. I restarted the failed tests and they passed so we are good to go.

**Comment by danielpost** [comment]
r251188-wpcom

**Pull request merged**